Projects/Cyber-Range-K8/K8_cluster_creation.md

```diff
@@ -0,0 +1,114 @@
+
+# Kubernetes Cluster Creation Notes
+
+## TOC
+- [Kubernetes Cluster Creation Notes](#kubernetes-cluster-creation-notes)
+  - [TOC](#toc)
+  - [Prerequisites](#prerequisites)
+  - [Steps](#steps)
+    - [Install Docker on Each Machine](#install-docker-on-each-machine)
+    - [Install Kubernetes Components on Each Machine](#install-kubernetes-components-on-each-machine)
+  - [Step 1: Initialize the Control Plane](#step-1-initialize-the-control-plane)
+    - [Step 1.1: Set Up kubeconfig for kubectl on Control Plane](#step-11-set-up-kubeconfig-for-kubectl-on-control-plane)
+    - [Step 1.2: Deploy a Pod Network(In This Case, Calico)](#step-12-deploy-a-pod-networkin-this-case-calico)
+      - [Why?](#why)
+  - [Step 2: Join Worker Nodes to the Cluster](#step-2-join-worker-nodes-to-the-cluster)
+  - [Step 3: Verify the Cluster](#step-3-verify-the-cluster)
+    - [Troubleshooting](#troubleshooting)
+  - [Notes](#notes)
+
+
+
+## Prerequisites
+1. Ensure all machines (control plane and worker nodes) are running a compatible Linux OS.
+   
+## Steps
+1. Install Docker or another container runtime on each machine.
+2. Install `kubeadm`, `kubelet`, and `kubectl` on each machine.
+
+### Install Docker on Each Machine
+```sh
+    sudo apt-get update
+    sudo apt-get install -y apt-transport-https ca-certificates curl software-properties-common
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+    sudo apt-get update
+    sudo apt-get install -y docker-ce
+    sudo systemctl enable docker
+    sudo systemctl start docker
+```
+
+### Install Kubernetes Components on Each Machine
+```sh
+    sudo apt-get update
+    sudo apt-get install -y apt-transport-https ca-certificates curl
+    sudo curl -fsSL https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+    echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee /etc/apt/sources.list.d/kubernetes.list
+    sudo apt-get update
+    sudo apt-get install -y kubelet kubeadm kubectl
+```
+
+## Step 1: Initialize the Control Plane
```

Comment:
What does this do?

I can guess based on the contents of the command.

What does it look like when this command succeeds? You can paste output or a screenshot.

Reply:
I do not have pictures, nor outputs to show at the moment. Will likely, not have any for some time.

What does what do?